modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js

```diff
@@ -116,19 +113,19 @@ const useFormSubmit = ({apiRef, containerRef}) => {
 					this
 				);
 
-				submitHandle = dom.on(form, 'submit', handleFormSubmitted);
+				form.addEventListener('submit', handleFormSubmitted);
 			}
-		}
 
-		return () => {
-			if (onHandle) {
-				onHandle.detach();
-			}
+			return () => {
+				if (onHandle) {
+					onHandle.detach();
+				}
 
-			if (submitHandle) {
-				submitHandle.removeListener();
-			}
-		};
+				if (form) {
+					form.removeEventListener('submit', handleFormSubmitted);
+				}
+			};
```

Comment:
Just incorporating by reference my comments from [the other PR](https://github.com/liferay-frontend/liferay-portal/pull/936#discussion_r601504168). There are a couple of code smells here — not your fault, @kresimir-coko, as they were pre-existing, but we still want to fix them.

Reply:
I'll tackle that and push

Reply:
> but we still want to fix them.

Funnily enough, that is a typo. I actually meant to write:

> but we still might want to fix them.

![oops](https://user-images.githubusercontent.com/7074/112488729-1c973780-8d7e-11eb-9089-d01fb6f71eaf.gif)

Reply:
I've force-pushed the changes, as they're on the same file @wincent.
This should do the trick, and I should do better next time, after zooming out from the `dom.on` line, it looked very smelly, like you noticed.